raspberry/bzzz/sensors/gnss.py

```diff
@@ -131,6 +134,25 @@ def __get_measurements_in_background_t(self, serial_path, baud):
         ser.close()
         return
 
+    def GNSS_altitude_Initilisation(self):
+        altitude_values_for_Initilisation = []
+        """
+        Returns the calibrated GNSS altitude based on the average 
+        altitude measured in the first minute of readings.
+        """
+        if not self.__calibrated:
+
+            for i in range(120):
+                time.sleep(1)
```

Comment:
This will take 2 minutes. Isn't this too much?

Reply:
This depends on the variance of the GNSS data. By gathering $N$ iid measurements the variance becomes $\sigma^2/N$.

Reply:
ok, we can test this by taking the drone outside with all correction data active and gathering data. we can then analyse the variance within the 1-2 minute mark to get the optimal time for gathering data